src/docs/arch-row.dox

```diff
@@ -1,12 +1,89 @@
+unsigned integer record id as keys, thus, column-stores implement a specialized version of a B-Tree
+where the key has a predictable length. In the case of column-stores, keys are implied given the
+B-Tree position.
+
+There are two types of column-stores in WiredTiger:
+- The fixed length column-store, as its name implies, the value has a fixed length, and furthermore
+the value is restricted to 1 and 8 bits in length. The bit length is specified when the column store
+is created. The fixed length column store has specialized use cases like bitmaps.
+- The variable length is a more general case which allows for values to have any length similarly to
+row-stores.
+
+@section In-memory and on-disk format
+
+Row-stores and column-stores have a specific in-memory and on-disk data structure. Please refer to
+the Data File Format section to learn more.
+
+@section Row-stores
+
+Row-stores are probably seen as the traditional database management system and they serve a general
+purpose. In WiredTiger, row-store keys are explicitly stored and they are usually duplicated within
+the same single file. In order to locate a specific record and pull together a complete row, a
+search algorithm is required. When reading and writing row-stores, the memory usage will depend on
+the definition of column groups required by the query.
+
+@section Variable length column-stores
+
+The main advantage of variable length column-stores is probably storage. Their keys are not stored
+but derived from the B-Tree location in the memory. In fact, there is a starting key on each page
+which is used to figure out the rest of the keys present on that same page. Column-stores values
+also present an advantage to further save more disk space as they can be written in more compact
+forms through encoding techniques. In WiredTiger, the run-length encoding is used to replace any
+sequence of the same value with a count and value indicator. However, this makes the access to a
+specific record more complicated, it is no longer possible to jump to a record in a leaf page.
+Finally, when reading and writing column-stores, only the columns required for any particular query
+are maintained in memory.
+
+@section Fixed length column-stores
+
+Fixed length column stores are very different internally and usually serve specific purposes as it
+is a bitmap. It makes fixed length column-stores efficient to retrieve a value at a given offset,
+the use of bloom filters is probably one of the best examples.
+
+@section Performance
+
+Generally, row-stores are faster for queries where all of the columns are required by every lookup
+because there is only a single set of internal pages to read into the cache and search.
+Column-stores are faster when most queries require only a subset of the columns because columns can
+be separated into multiple files and only the columns being returned need to be present in the
```

Comment:
Per my comment above - it's possible to split a table into multiple files using a combination of row stores and column stores, so this benefit can be gained regardless of access method choice.

It's more common to split when using column stores, since the space efficiency gains are greatest when there are many repeated values, and that is less likely across all the columns in a row.

It's worth noting in this section that MongoDB doesn't use the rich schema being described. It stores a blob of data in a row store and manages encoding and decoding itself.

Reply:
I reworked this section, inviting the reader to learn more about column groups, hopefully at the right time. Let me know if that sounds good to you, see c4dcd44e65a917bcf879977b468353a8338bb55d

Reply:
Looks great now - thanks!